language-server/dm/documentManager.ml

```diff
@@ -739,6 +739,57 @@ let hover st pos =
   let lconstr = Procq.Constr.lconstr
 [%%endif]
 
+
+[%%if coq = "8.18" || coq = "8.19" || coq = "8.20"]
+let jump_to_definition _  _ = None
+[%%else]
+let jump_to_definition st pos =
+  let raw_doc = Document.raw_document st.document in
+  let loc = RawDocument.loc_of_position raw_doc pos in
+  let opattern = RawDocument.word_at_position raw_doc pos in
+  match opattern with
+  | None -> log "jumpToDef: no word found at cursor"; None
+  | Some pattern ->
+    log ("jumpToDef: found word at cursor: \"" ^ pattern ^ "\"");
+    match st.observe_id with
+    | Top -> log "jumpToDef with no context"; None
+    | (Id id) ->
+      match Document.get_sentence st.document id with
+      | None -> log "jumpToDef: observe_id does not exist"; None
+      | Some { stop } ->
+        let o_pos = RawDocument.position_of_loc raw_doc stop in
+        match get_context st o_pos with
+        | None -> log "No context found"; None
+        | Some _ ->
```

Comment:
this is fishy, why we check for a context and then don't use it?
was this preliminary code to use the proof context when available?
At least it deserves a comment

Reply:
Yeah, I think its just a way to check that we are indeed in an executed context.
If I just jump to def with no execution state than I just get an uncaught Not_found exception. But I suppose I should probably catch the exception and return None in that case. Or check that the sentence at the cursor is executed ?